agent/ecr/client.go

```diff
@@ -43,29 +40,18 @@ type ECRSDK interface {
 }
 
 type ecrClient struct {
-	sdkClient  ECRSDK
-	tokenCache async.Cache
+	sdkClient ECRSDK
 }
 
-func NewECRClient(sdkClient ECRSDK, tokenCache async.Cache) ECRClient {
+// NewECRClient creates a ecrclient used to get docker auth from ecr
+func NewECRClient(sdkClient ECRSDK) ECRClient {
 	return &ecrClient{
-		sdkClient:  sdkClient,
-		tokenCache: tokenCache,
+		sdkClient: sdkClient,
 	}
 }
 
+// GetAuthorizationToken calls the ecr api to get the docker auth for the specified registry
```

Comment:
GetAuthorizationToken calls the ECR API to get the docker auth for the specified registry.